src/pint/models/noise_model.py

```diff
@@ -556,6 +561,125 @@ def pl_dm_cov_matrix(self, toas):
         return np.dot(Fmat * phi[None, :], Fmat.T)
 
 
+class PLSWNoise(NoiseComponent):
+    """Model of solar wind DM variations as radio frequency-dependent noise with a
+    power-law spectrum.
+
+    Commonly used as perturbations on top of a deterministic solar wind model.
+
+
+    Parameters supported:
+
+    .. paramtable::
+        :class: pint.models.noise_model.PLSWNoise
+
+    Note
+    ----
+    Ref: Lentati et al. 2014, MNRAS 437(3), 3004-3023.
+    See also: Hazboun et al. 2019 and Susurla et al. 2024.
+
+    """
+
+    register = True
+    category = "pl_SW_noise"
+
+    introduces_correlated_errors = True
+    is_time_correlated = True
+
+    def __init__(
+        self,
+    ):
+        super().__init__()
+
+        self.add_param(
+            floatParameter(
+                name="TNSWAMP",
+                units="",
+                aliases=[],
+                description="Amplitude of power-law SW DM noise in tempo2 format",
+                convert_tcb2tdb=False,
+            )
+        )
+        self.add_param(
+            floatParameter(
+                name="TNSWGAM",
+                units="",
+                aliases=[],
+                description="Spectral index of power-law "
+                "SW DM noise in tempo2 format",
+                convert_tcb2tdb=False,
+            )
+        )
+        self.add_param(
+            floatParameter(
+                name="TNSWC",
+                units="",
+                aliases=[],
+                description="Number of SW DM noise frequencies.",
+                convert_tcb2tdb=False,
+            )
+        )
+
+        self.covariance_matrix_funcs += [self.pl_sw_cov_matrix]
+        self.basis_funcs += [self.pl_sw_basis_weight_pair]
+
+    def get_pl_vals(self):
+        nf = int(self.TNSWC.value) if self.TNSWC.value is not None else 30
+        amp, gam = 10**self.TNSWAMP.value, self.TNSWGAM.value
+        return (amp, gam, nf)
+
+    def get_noise_basis(self, toas, planetssb, sunssb, pos_t):
+        """Return a Fourier design matrix for SW DM noise.
+
+        See the documentation for pl_sw_basis_weight_pair function for details."""
+
+        tbl = toas.table
+        t = (tbl["tdbld"].quantity * u.day).to(u.s).value
+        freqs = self._parent.barycentric_radio_freq(toas).to(u.MHz)
+        nf = self.get_pl_vals()[2]
+        # get the acrhomatic Fourier design matrix
+        Fmat = create_fourier_design_matrix(t, nf)
+        # then get the solar wind chromatic part to multiply by
```

Comment:
I see that this Fmat  only includes the sin and cos components with fundamental frequency 1/Tspan. This will not fit the data well if there is a linear or quadratic variation in NE_SW. I suggest adding derivatives of NE_SW in `SolarWindDispersion` to take care of this.

Reply:
See #1859

Reply:
thanks for coding this in!!
i will try to get all of the tests passing on this PR so that we can get these both merged at the next git-flowers meeting

Reply:
i have thought about this more as well. it is great to have these polynomial fits in PINT, however, there are 2 reasons that i think they will not always be required with this model. 
1) the spectra will not always be very “red”. infact, i think Sai found that they are often blue. so I think the low frequency power is less important. 
2) DM1 and DM2 will probably absorb much of this since the solar wind is very covariant w DM at low frequencies. i think Caterina shows this in one of her papers.

Reply:
Ok... In any case I have implemented the NE_SW derivatives in another PR which is already merged. The user can use them if needed, but are not necessary.